lib/active_record/connection_adapters/nulldb_adapter/core.rb

```diff
@@ -164,22 +164,28 @@ def execute(statement, name = nil)
   end
 
   def exec_query(statement, name = 'SQL', binds = [], options = {})
+    internal_exec_query(statement, name, binds, **options)
+  end
+
+  def internal_exec_query(statement, name = 'SQL', binds = [], prepare: false, async: false)
     self.execution_log << Statement.new(entry_point, statement)
     EmptyResult.new
   end
 
-  def select_rows(statement, name = nil, binds = [])
+  def select_rows(statement, name = nil, binds = [], async: false)
     [].tap do
       self.execution_log << Statement.new(entry_point, statement)
     end
   end
 
   def insert(statement, name = nil, primary_key = nil, object_id = nil, sequence_name = nil, binds = [])
-    (object_id || next_unique_id).tap do
+    result = (object_id || next_unique_id).tap do
       with_entry_point(:insert) do
         super(statement, name, primary_key, object_id, sequence_name)
       end
     end
+
+    Array.wrap(result)
```

Comment:
Saw this PR after doing https://github.com/nulldb/nulldb/pull/125

I think for this method we should only return an array if the `returning` prop is provided. This matches how other adapters work.

Reply:
Thanks for the extra insight, fixed this.